src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -108,12 +119,28 @@ private void ifUsernameAlreadyExistsThrowException(String username) {
     }
   }
 
-  private User findUserToActivateOrThrowException(UUID userId,
-      UUID activationToken) {
-    return userRepository.findByUserIdAndActivationTokenAndActivatedIsFalse(userId,
-            activationToken)
-        .orElseThrow(
-            () -> new IllegalArgumentException("Token is invalid or user is already activated"));
+  private User findUserToActivateOrThrowException(UUID userId, UUID activationTokenUUID) {
+    Token activationToken = tokenRepository.findByTokenId(activationTokenUUID)
+            .orElseThrow(() -> new IllegalArgumentException("Token is invalid or user is already activated"));
+      return activationToken.getUser();
+  }
+
+  private UUID generateActivationToken(User user){
+    Token activationToken = new Token();
+    activationToken.builder()
+            .user(user)
+            .build();
```

Comment:
O llamamos al constructor o llamamos al builder

```suggestion
    Token activationToken =    activationToken.builder()
            .user(user)
            .build();

```

Ambas cosas tienen el mismo objetivo